contracts/v6/OraclePrice.sol

```diff
@@ -0,0 +1,245 @@
+// SPDX-License-Identifier: Unlicensed
+pragma solidity =0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "./interface/KeeperCompatibleInterface.sol";
+import "./interface/IOmsPolicy.sol";
+import "./interface/EACAggregatorProxy.sol";
+import "./library/SafeMath.sol";
+import "./common/Ownable.sol";
+
+contract OraclePrice is Ownable, KeeperCompatibleInterface {
+    using SafeMath for uint256;
+
+    struct PriceLog {
+        int256 lastUpdatedPrice;
+    }
+
+    struct AverageLog {
+        int256 averageMovement;
+        int256 referenceRate;
+    }
+
+    struct OracleInfo {
+        address oracleAddress;
```

Comment:
Thanks for that mate.. it's all about the way things are written to the block chain and since each storage slot is 32 bytes. 

For us, the OracleInfo struct will take 3 slots at the moment since the first 2 variables fit in one slot.

slot 1: address (20 bytes) + bool (1 byte)
slot 2: bytes32 (32 bytes)
slot 3: int256 (32 bytes)

If we had the boolean as the 3rd or 4th value in the struct, this would lead to 4 slots being used as the address and the bytes32 variables wouldn't fit in one slot. 

Inefficient code:
slot 1: address (20 bytes)
slot 2: bytes32 (32 bytes)
slot 3: bool (1 byte)
slot 4: int256 (32 bytes)

Reply:
Yeah ok makes total sense. I had in my head that since the first 21 bytes aren't big enough it without have to split the 32 across 2 slots. Don't mind me haha